lib/elixir/lib/kernel/typespec.ex

```diff
@@ -861,4 +883,25 @@ defmodule Kernel.Typespec do
   end
 
   defp variable(expr), do: expr
+
+  defp update_used_vars({global_state, local_state}, var_name) do
+    case Keyword.fetch(local_state.used_vars, var_name) do
+      {:ok, :used_once} ->
+        used_vars = Keyword.put(local_state.used_vars, var_name, :used_multiple)
```

Comment:
Why not a map for `used_vars`?

Reply:
I don't think we should worry about this. Those warnings are unlikely and whatever comes first will be enough. Also, I don't think Keyword.put guarantees order.